Make navigation OpenCV4 friendly

Hi @randaz81 

As of now, `navigation` doesn't seem Focal ready, as per the following compilation error:

```sh
In file included from /usr/local/src/robot/navigation/src/navigationGUI/navGui.h:48,
                 from /usr/local/src/robot/navigation/src/navigationGUI/main.cpp:30:
/usr/local/src/robot/navigation/src/navigationGUI/map.h:30:10: fatal error: cv.h: No such file or directory
   30 | #include <cv.h>
      |          ^~~~~~
```

It would be great to refactor modules needing OpenCV so that they'll be compatible with the version 4.0 of the library.

cc @vvasco

--- a/src/navigationDevices/robotPathPlannerDevice/map.cpp
+++ b/src/navigationDevices/robotPathPlannerDevice/map.cpp
@@ -23,8 +23,6 @@
 #include <yarp/dev/Map2DLocation.h>
 #include <string>
 #include <math.h>
-#include <cv.h>
-#include <highgui.h> 
 
 #include "map.h"
 #include "aStar.h"

--- a/src/navigationDevices/robotPathPlannerDevice/map.h
+++ b/src/navigationDevices/robotPathPlannerDevice/map.h
@@ -27,8 +27,6 @@
 #include <yarp/sig/ImageDraw.h>
 #include <yarp/dev/MapGrid2D.h>
 #include <string>
-#include <cv.h>
-#include <highgui.h> 
 #include <queue>
 
 using namespace std;

--- a/src/navigationDevices/robotPathPlannerDevice/pathPlannerCtrl.cpp
+++ b/src/navigationDevices/robotPathPlannerDevice/pathPlannerCtrl.cpp
@@ -34,8 +34,7 @@
 #define _USE_MATH_DEFINES
 #include <math.h>
 
-#include <cv.h>
-#include <highgui.h> 
+#include <opencv2/imgproc/imgproc_c.h>
 
 #include "pathPlannerCtrl.h"
 #include "pathPlannerCtrlHelpers.h"

--- a/src/navigationGUI/map.cpp
+++ b/src/navigationGUI/map.cpp
@@ -33,8 +33,6 @@
 #include <yarp/dev/Map2DLocation.h>
 #include <string>
 #include <math.h>
-#include <cv.h>
-#include <highgui.h> 
 
 #include "map.h"
 

--- a/src/navigationGUI/map.h
+++ b/src/navigationGUI/map.h
@@ -27,8 +27,7 @@
 #include <yarp/sig/ImageDraw.h>
 #include <yarp/dev/MapGrid2D.h>
 #include <string>
-#include <cv.h>
-#include <highgui.h> 
+#include <opencv2/imgproc/imgproc_c.h>
 #include <queue>
 
 using namespace std;

--- a/src/navigationGUI/navGui.cpp
+++ b/src/navigationGUI/navGui.cpp
@@ -36,16 +36,6 @@
 #define _USE_MATH_DEFINES
 #include <math.h>
 
-#include <opencv2/core/version.hpp>
-#if CV_MAJOR_VERSION >= 3
-#include <opencv2/highgui/highgui_c.h>
-#else
-#include <cv.h>
-#include <highgui.h>
-#endif
-
-#include <highgui.h> 
-
 #include "navGui.h"
 
 using namespace std;

